Benqi-Agents/Qi-Transfers/src/data.fetcher.ts

```diff
@@ -0,0 +1,20 @@
+import { Contract, providers, BigNumber } from "ethers";
+import { benqiInterface } from "./utils";
+
+export default class DataFetcher {
+  readonly qiTokenAddress: string;
+  readonly provider: providers.Provider;
+  private qiTokenContract: Contract;
+
+  constructor(qiTokenAddress: string, provider: providers.Provider) {
+    this.qiTokenAddress = qiTokenAddress;
+    this.provider = provider;
+    this.qiTokenContract = new Contract(qiTokenAddress, benqiInterface, provider);
+  }
+
+  // return the number of tokens held by the account
+  public async getBalance(account: string): Promise<BigNumber> {
+    const balance: BigNumber = await this.qiTokenContract.balanceOf(account, { blockTag: "latest" });

```

Comment:
why are you fetching from `latest` instead of an specific block?

Reply:
because `balanceOf` function in the contract only takes an `account` as a parameter. and I thought whenever it is called, the most updated balance of an account is important. should it be for the specific block in which agent is running on at that time?

Reply:
`blockTag` let you specify the block from where you want to take the data. That means that you will be calling `balanceOf` in that specific block, so the value returned will be the one the address had in that block instead of the most updated balance. The the answer to the question is yes.